core/src/main/scala/filodb.core/query/RangeVector.scala

```diff
@@ -61,6 +61,11 @@ final case class CustomRangeVectorKey(labelValues: Map[UTF8Str, UTF8Str]) extend
   val sourceShards: Seq[Int] = Nil
 }
 
+final case class CustomRangeVectorKeyWithShards(labelValues: Map[UTF8Str, UTF8Str], sourceShards: Seq[Int])
```

Comment:
Why was this needed? I think you can simply use CustomRangeVectorKey

Reply:
Removed CustomRangeVectorKeyWithShards and modified CustomRangeVectorKey to have shards as input

Reply:
Ah - what I meant is that shards should not be part of the key. Why is this needed in the first place ?

Reply:
I could not just update labelValues of RangeVectorKey since it is a val. So I created new RangeVectorKey with the updated labels. If I use the original CustomRangeVectorKey, sourceShards will be nil.
So in labelReplaceImpl I create updated RangeVectorKey using: CustomRangeVectorKey(rangeVectorKey.labelValues.
         updated(ZeroCopyUTF8String(dstLabel), ZeroCopyUTF8String(labelReplaceValue)), rangeVectorKey.sourceShards)